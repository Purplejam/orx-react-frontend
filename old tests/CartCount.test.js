import {render, screen} from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
	it('Renders', () => {
		render(<CartCount count={10}/>)
	})

	it('Matches the snapshot', () => {
		const {container} = render(<CartCount count={11}/>);
		expect(container).toMatchSnapshot();
	})

	it('updates via props', async () => {
		const {container, rerender} = render(<CartCount count={11}/>);
		expect(container).toHaveTextContent('11');

		//update props
		rerender(<CartCount count={12}/>);
		//wait
		await wait(400);
		expect(container).toHaveTextContent('12');
		expect(container).toMatchSnapshot();
	})
})