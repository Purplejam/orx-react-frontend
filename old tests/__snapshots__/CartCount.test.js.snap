// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches the snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <p
        class="CartCount__Dot-sc-xxvp4g-0 glPYDY count"
      >
        11
      </p>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <p
        class="CartCount__Dot-sc-xxvp4g-0 glPYDY count count-enter-done"
      >
        12
      </p>
    </div>
  </span>
</div>
`;
